lib/gakubuchi/template.rb

```diff
@@ -2,54 +2,73 @@ module Gakubuchi
   class Template
     extend ::Forwardable
 
-    attr_reader :pathname
-    def_delegators :@pathname, :hash
+    attr_reader :source_path
+    def_delegator :source_path, :hash
 
     %w(== === eql?).each do |method_name|
       define_method(method_name) do |other|
-        self.class == other.class && @pathname.__send__(method_name, other.pathname)
+        self.class == other.class && source_path.public_send(method_name, other.source_path)
       end
     end
 
     def self.all
-      ::Dir.glob(root.join('**/*.html*')).map { |path| new(path) }
+      ::Dir.glob(root.join('**/*.html*')).map { |source_path| new(source_path) }
     end
 
     def self.root
       ::Rails.root.join('app/assets', ::Gakubuchi.configuration.template_directory)
     end
 
-    def initialize(path)
-      @pathname = ::Pathname.new(path)
+    def initialize(source_path)
+      path = ::Pathname.new(source_path)
+
+      case
+      when path.relative?
+        @source_path = self.class.root.join(path)
+      when path.ascend { |parent| break true if parent == self.class.root }
+        @source_path = path
+      else
+        fail "source path must include #{self.class.root} when it is absolute"
+      end
     end
 
-    def destination_pathname
-      dirname = relative_pathname.dirname
-      ::Rails.public_path.join(dirname, "#{relative_pathname.basename(extname)}.html")
+    def destination_path
+      ::Rails.public_path.join(logical_path)
+    end
+
+    def digest_path
```

Comment:
[IMO]
Related to [this comment](https://github.com/yasaichi/gakubuchi/pull/1/files#r49253718), you should implement this method as follows.

```ruby
def digest_path_in_context(context)
  # do something
end
```